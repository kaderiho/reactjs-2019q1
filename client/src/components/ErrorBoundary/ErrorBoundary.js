import React from 'react';

class ErrorBoundary extends React.Component {
    state = {
        hasError: false,
        errorInfo: null,
        error: null
    };

    componentDidCatch(error, errorInfo) {
        this.setState({
            hasError: true,
            errorInfo,
            error
        });
    }

    render() {
        const { hasError, errorInfo, error } = this.state;
        const { render, children } = this.props;

        if (hasError) {
            return render(errorInfo, error);
        }

        return children;
    }
}

export default ErrorBoundary;
