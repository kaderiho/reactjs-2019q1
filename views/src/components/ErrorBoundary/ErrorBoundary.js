"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var ErrorBoundary =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(ErrorBoundary, _React$Component);

  function ErrorBoundary() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, ErrorBoundary);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(ErrorBoundary)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      hasError: false,
      errorInfo: null,
      error: null
    });
    return _this;
  }

  (0, _createClass2.default)(ErrorBoundary, [{
    key: "componentDidCatch",
    value: function componentDidCatch(error, errorInfo) {
      this.setState({
        hasError: true,
        errorInfo: errorInfo,
        error: error
      });
    }
  }, {
    key: "render",
    value: function render() {
      if (this.state.hasError) {
        return this.props.render(this.state.errorInfo, this.state.error);
      }

      return this.props.children;
    }
  }]);
  return ErrorBoundary;
}(_react.default.Component);

var _default = ErrorBoundary;
exports.default = _default;