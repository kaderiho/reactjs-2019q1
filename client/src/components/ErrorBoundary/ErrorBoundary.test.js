import React from 'react';
import { shallow, mount } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

/**
 * Tests
 * 
 * - To match snapshot
 * - Show error message in case of children error happened
 */

describe('ErrorBoundary component', () => {
    it('Should render correctly with default props', () => {
        const mockComponent = <div>My mock element</div>
        const wrapper = shallow(<ErrorBoundary>{mockComponent}</ErrorBoundary>);

        expect(wrapper).toMatchSnapshot();
    });

    it('Should render error message', () => {
        const errorRenderMessage = <div>Oop! Something went wrong.</div>
        const FailedComponent = () => {
            throw new Error('Error happened!');
            return;
        };
        const wrapper = mount(
        <ErrorBoundary 
            render={() => errorRenderMessage }>
            <FailedComponent/>
        </ErrorBoundary>);

        expect(wrapper).toMatchSnapshot();
    })
})
