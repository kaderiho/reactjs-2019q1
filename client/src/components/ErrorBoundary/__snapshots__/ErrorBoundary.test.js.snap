// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component Should render correctly with default props 1`] = `
<div>
  My mock element
</div>
`;

exports[`ErrorBoundary component Should render error message 1`] = `
<ErrorBoundary
  render={[Function]}
>
  <div>
    Oop! Something went wrong.
  </div>
</ErrorBoundary>
`;
